File: src/main/java/com/felipe/booking/app/entity/BookingLockEntity.java
package com.felipe.booking.app.entity;

import com.felipe.booking.domain.model.BookingLock;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Version;
import org.springframework.data.mongodb.core.mapping.Document;

@Document("booking_lock")
public class BookingLockEntity {
    @Id
    private String roomId;
    @Version
    private Long version;

    public BookingLock toDomain() {
        return new BookingLock(roomId, version);
    }
}

File: src/main/kotlin/com/felipe/booking/alten/app/entity/BookingLockEntity.kt
package com.felipe.booking.alten.app.entity

import com.felipe.booking.alten.domain.model.BookingLock
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Version
import org.springframework.data.mongodb.core.mapping.Document

@Document("booking_lock")
data class BookingLockEntity(
    @Id val roomId: String,
    @Version val version: Long?
) {
    fun toDomain() =
        BookingLock(
            roomId = roomId,
            version = version!!
        )
}